core/src/main/java/io/questdb/griffin/DatabaseSnapshotAgent.java

```diff
@@ -262,27 +339,43 @@ public void prepareSnapshot(SqlExecutionContext executionContext) throws SqlExce
                 final int tableNameIndex = factory.getMetadata().getColumnIndex(TableListRecordCursorFactory.TABLE_NAME_COLUMN);
                 try (RecordCursor cursor = factory.getCursor(executionContext)) {
                     final Record record = cursor.getRecord();
+
+                    // Suspend the WalPurgeJob
+                    if (walPurgeJobRunLock != null) {
+                        final long timeout = configuration.getCircuitBreakerConfiguration().getTimeout();
```

Comment:
Do you mean replace `lock.tryLock(timeout) `with something like:

```
while(true) {
    if (lock.tryLock()) {
        break;
    }
    braker.statefulThrowExceptionIfTrippedNoThrottle();
}
``` ?
It's actually a busy-waiting which I'd like to avoid.

Reply:
```
while(true) {
    if (lock.tryLock(configuration.getCircuitBreakerConfiguration().getTimeout())) {
        break;
    }
    braker.statefulThrowExceptionIfTrippedNoThrottle();
}
```

Timeout can be extended per request so the config value can be below what user's wants to use.
I would make sense say in our cloud to run it with bigger timeout than default.